pdfiumandroid/src/main/java/io/legere/pdfiumandroid/util/PdfiumNativeSourceBridge.kt

```diff
@@ -0,0 +1,28 @@
+package io.legere.pdfiumandroid.util
+
+import io.legere.pdfiumandroid.BuildConfig
+import io.legere.pdfiumandroid.PdfiumSource
+
+internal class PdfiumNativeSourceBridge(private val source: PdfiumSource) {
+
+    private var buffer: ByteArray? = null
+
+    fun read(position: Long, size: Long): Int =
+        try {
+            require(size <= Int.MAX_VALUE) { "size is too large" }
+            val trimmedSize = size.toInt()
+
+            var buffer = buffer
+            if (buffer == null || buffer.size < size) {
+                buffer = ByteArray(trimmedSize).also { this.buffer = it }
+            }
+
+            val bytesRead = source.read(position, buffer, trimmedSize)
+            if (bytesRead <= 0) 0 else bytesRead // Pdfium expects 0 for error while Java/Kotlin usually return a negative value
+        } catch (t: Throwable) { // This is to prevent the exception to go to the native code level
```

Comment:
If for some reason `source.read()` throws an exception, we're in trouble with JNI. We need to check for pending exception and process it accordingly. We can do this in JNI, but it'd require much more code to write and maintain. This is simple and it does everything the JNI code would do: log exception and return 0 to Pdfium.

Reply:
That's fine. But detekt will complain, please add @Suppress("TooGenericExceptionCaught") above the method.